sycl/source/detail/graph_impl.cpp

```diff
@@ -1775,6 +1844,208 @@ void dynamic_parameter_base::updateAccessor(
   impl->updateAccessor(Acc);
 }
 
+void dynamic_parameter_impl::updateValue(const raw_kernel_arg *NewRawValue,
+                                         size_t Size) {
+  // Number of bytes is taken from member of raw_kernel_arg object rather
+  // than using the size parameter which represents sizeof(raw_kernel_arg).
+  std::ignore = Size;
+  size_t RawArgSize = NewRawValue->MArgSize;
+  const void *RawArgData = NewRawValue->MArgData;
+
+  for (auto &[NodeWeak, ArgIndex] : MNodes) {
+    auto NodeShared = NodeWeak.lock();
+    if (NodeShared) {
+      dynamic_parameter_impl::updateCGArgValue(
+          NodeShared->MCommandGroup, ArgIndex, RawArgData, RawArgSize);
+    }
+  }
+
+  for (auto &DynCGInfo : MDynCGs) {
+    auto DynCG = DynCGInfo.DynCG.lock();
+    if (DynCG) {
+      auto &CG = DynCG->MKernels[DynCGInfo.CGIndex];
+      dynamic_parameter_impl::updateCGArgValue(CG, DynCGInfo.ArgIndex,
+                                               RawArgData, RawArgSize);
+    }
```

Comment:
Just to clarify my own understanding: If multiple CGs inside a single `dynamic_command_group` use the same dynamic parameter, they will be stored as separate entries in `MDynCGs`. Is that correct?

Reply:
Yes that's correct, so the dynamic parameter impl class stores:
```cpp
/// Struct detailing an instance of the usage of the dynamic parameter in a
/// dynamic CG.
  struct DynamicCGInfo {
  /// Dynamic command-group that uses this dynamic parameter.
  std::weak_ptr<dynamic_command_group_impl> DynCG;
  /// Index of the CG in the Dynamic CG that uses this dynamic parameter.
  size_t CGIndex;
  /// The arg index in the kernel the dynamic parameter is used.
  int ArgIndex;
};

std::vector<DynamicCGInfo> MDynCGs;  
```

I did consider doing it with a per dynamic cg container, but wasn't sure if it would overly complicate the readability. e.g
```cpp
std::map<std::weak_ptr<dynamic_command_group_impl>, std::vector<CGIndex, ArgIndex>> MDynCGs
```